tests/acceptance/features/bootstrap/SpacesContext.php

```diff
@@ -3329,4 +3329,54 @@ public function userDownloadsTheSpaceUsingTheWebdavApi(string $user, string $spa
 			)
 		);
 	}
+
+	/**
+	 * @When user :user gets the content of file :fileName using WebDav API
+	 * @When user :user gets the content of file :fileName inside space :spaceName using WebDav API
+	 * @When user :user gets the content of file :fileName shared by user :sharer using WebDav API
+	 * @When user :user gets the content of file :fileName inside space :spaceName shared by user :sharer using WebDav API
+	 *
+	 * @param string $user
+	 * @param string $fileName
+	 * @param string|null $spaceName
+	 * @param string|null $sharer
+	 *
+	 * @return void
+	 */
+	public function userGetsTheContentOfFileUsingWebDavAPI(string $user, string $fileName, ?string $spaceName = null, ?string $sharer = null): void {
+		$user = $sharer ?? $user;
+		$user = $this->featureContext->getActualUsername($user);
+
+		if ($spaceName === null) {
+			$fileId = $this->featureContext->getFileIdForPath($user, $fileName);
+		} else {
+			$fileId = $this->getFileId($user, $spaceName, $fileName);
+		}
+
+		$this->featureContext->setResponse(
+			WebDavHelper::makeDavRequest(
+				$this->featureContext->getBaseUrl(),
+				$user,
+				$this->featureContext->getPasswordForUser($user),
+				'GET',
+				$fileId,
```

Comment:
I think, this `$fileId` should be enough. I don't think we have to edit `getDavPath` method. But I may be wrong here :thinking:

Reply:
yeah `getDavPath` method currently doesn't return the expected dav path `/remote.php/dav/spaces/`. That's also the reason I added `$type='fileId'` to get the required dav path.

Reply:
Ah, now I get it. The endpoints must be `/spaces/<fileId>` not `/spaces/<spaceId>/<fileId>`.